src/parser/semantic_check.c

```diff
@@ -11424,7 +11424,7 @@ pt_assignment_compatible (PARSER_CONTEXT * parser, PT_NODE * lhs, PT_NODE * rhs)
 		  if (rhs->node_type != PT_HOST_VAR)
 		    {
 		      d = tp_domain_resolve_default (lhs_dbtype);
-		      if (PT_IS_PARAMETERIZED_TYPE (lhs->type_enum))
```

Comment:
checking `lhs->data_type != NULL` is not needed. `lhs->data_type` is not accessed in the if-block.

Reply:
The following code snippet is from the earlier part of this routine, the **lhs->data_type** is checked if it's null. I think the checking is needed.
```
if (lhs->data_type)
	{
	  sci.prec = lhs->data_type->info.data_type.precision;
	  sci.scale = lhs->data_type->info.data_type.dec_precision;
	  if (lhs->type_enum == PT_TYPE_JSON && lhs->data_type->info.data_type.json_schema != NULL)
	    {
	      sci.force_cast = true;
	    }
	}
```

Reply:
Ok, I agree